mm2src/coins/utxo/bchd_grpc.rs

```diff
@@ -79,11 +80,27 @@ pub struct ValidateSlpUtxosErr {
     kind: ValidateSlpUtxosErrKind,
 }
 
+impl From<ValidateSlpUtxosErr> for ValidatePaymentError {
+    fn from(err: ValidateSlpUtxosErr) -> Self {
+        match err.kind {
+            ValidateSlpUtxosErrKind::MultiReqErr(_) => Self::Transport(err.to_string()),
+            ValidateSlpUtxosErrKind::InvalidSlpTxData(_) => Self::WrongPaymentTx(err.to_string()),
+            _ => Self::InvalidRpcResponse(err.to_string()),
+        }
+    }
+}
+
 impl From<GrpcWebMultiUrlReqErr> for ValidateSlpUtxosErr {
     fn from(err: GrpcWebMultiUrlReqErr) -> Self {
-        ValidateSlpUtxosErr {
-            to_url: err.to_url.clone(),
-            kind: ValidateSlpUtxosErrKind::MultiReqErr(err),
+        match err.err {
+            PostGrpcWebErr::DecodeBody(e) => ValidateSlpUtxosErr {
```

Comment:
I guess https://github.com/KomodoPlatform/atomicDEX-API/pull/1444#issuecomment-1243832452 is related to this code?

Reply:
yes right.

Reply:
@borngraced 
I tried to roll back this diff to 
```rust
impl From<GrpcWebMultiUrlReqErr> for ValidateSlpUtxosErr {
    fn from(err: GrpcWebMultiUrlReqErr) -> Self {
        ValidateSlpUtxosErr {
            to_url: err.to_url.clone(),
            kind: ValidateSlpUtxosErrKind::MultiReqErr(err),
        }
    }
}
```

And got `test_validate_htlc_invalid_slp_utxo` failure with `Unexpected err Transport("Error Error DecodeBody(\"PayloadTooShort\")`. In debugger, I've noticed that response body is empty in `decode_body`. I might have false memories, but I think I encountered it during SLP integration. For some reason, BCHD responds with empty payload in case of invalid UTXO when grpc-web is used. Though there was a more meaningful response when using standard gRPC HTTP2 TLS transport (you can try it using tonic).

So, to handle this edge case (which actually looks like BCHD bug), I would add `PayloadTooShort` explicitly to `PostGrpcWebErr`, and map only this variant to `ValidateSlpUtxosErr::InvalidSlpTxData`. It's worth adding a comment describing this behavior in details.

PS I expect you to provide more of such context in the future. It took quite some time for me to understand to what code the initial question is *really* related and then test it myself.

cc @sergeyboyko0791

Reply:
Alright noted will work on it and take note of the last paragraph. thanks